master/internal/core_experiment.go

```diff
@@ -388,6 +389,11 @@ func (m *Master) parseCreateExperiment(ctx context.Context, req *apiv1.CreateExp
 
 	taskSpec.Project = config.Project()
 	taskSpec.Workspace = config.Workspace()
+	if p.Id > 1 {
+		// use project and workspace name from command
+		taskSpec.Project = p.Name
+		taskSpec.Workspace = workspaceModel.Name
+	}
```

Comment:
```suggestion
	taskSpec.Project = p.Name
	taskSpec.Workspace = workspaceModel.Name
```

Same point as commands, we should always set a Workspace name for when we add the label.